### PR TITLE
[KYUUBI #1501] Introduce operationsResource

### PR DESCRIPTION
**Pull request by simon824**
### _Why are the changes needed?_
#1501
Introduce operationsResource
mv parseSessionHandle() to SessionHandle
mv parseOperationHandle() to OperationHandle

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by simon824** [comment]
cc @yanghua @yaooqinn

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON)),
+    description =
+      "Get an operation detail with a given session identifier and operation identifier")
+  @GET
+  @Path("{operationHandle}")
+  def getOperationHandle(
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:43]: getOperationDetail

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala [hunk 2]
```diff
@@ -137,4 +142,23 @@ abstract class OperationManager(name: String) extends AbstractService(name) {
...
   }
+
+  def parseOperationHandle(operationHandleStr: String): OperationHandle = {
+    try {
+      val operationHandleParts = operationHandleStr.split("\\|")
+      require(
+        operationHandleParts.size == 4,
+        s"Expected 4 parameters but found ${operationHandleParts.size}.")
+
+      val handleIdentifier = new HandleIdentifier(
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala:153]: let have a `HandleIdentifier.apply(publicID: String, secretId: String): HandleIdentifier`

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala [hunk 3]
```diff
@@ -137,4 +142,23 @@ abstract class OperationManager(name: String) extends AbstractService(name) {
       }
     }
   }
+
+  def parseOperationHandle(operationHandleStr: String): OperationHandle = {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala:146]: lets move it to `object OperationHandle`

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 4]
```diff
@@ -283,4 +286,22 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
       timeoutChecker.scheduleWithFixedDelay(checkTask, interval, interval, TimeUnit.MILLISECONDS)
     }
   }
+
+  def parseSessionHandle(sessionHandleStr: String): SessionHandle = {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:290]: let's move it to `object SessionHandle`

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 5]
```diff
@@ -283,4 +286,22 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
...
+    try {
+      val sessionHandleParts = sessionHandleStr.split("\\|")
+      require(
+        sessionHandleParts.size == 3,
+        s"Expected 4 parameters but found ${sessionHandleParts.size}.")
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:295]: Expected 4 -> 3

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      @PathParam("operationHandle") operationHandleStr: String): OperationDetail = {
+    try {
+      val operationManager = backendService.sessionManager.operationManager
+      val operationHandle = operationManager.parseOperationHandle(operationHandleStr)
+      val operation = operationManager.getOperation(operationHandle)
+      OperationDetail(operation.shouldRunAsync, operation.isTimedOut, operation.getStatus)
+    } catch {
+      case NonFatal(_) =>
+        throw new NotFoundException(s"Error closing an operation")
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:52]: 'Error closing an operation'?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 7]
```diff
@@ -283,4 +286,22 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
...
+
+      val handleIdentifier = new HandleIdentifier(
+        UUID.fromString(sessionHandleParts(0)),
+        UUID.fromString(sessionHandleParts(1)))
+      val protocolVersion = TProtocolVersion.findByValue(sessionHandleParts(2).toInt)
+      new SessionHandle(handleIdentifier, protocolVersion)
+    } catch {
+      case NonFatal(_) =>
+        throw KyuubiSQLException(s"Invalid $sessionHandleStr")
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:304]: Does it seem to hide the root cause here? Shall we remove the entire `try-catch` or wrap for reason

**Review by yaooqinn — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**